File: pset6/readability.py
from cs50 import get_string
from re import compile

# input text
text = get_string("Text: ")

# count letters, words, and sentences
letter = 0
word = 1
sentence = 0
l = compile('[a-zA-Z]')

for n in text:
    if l.match(n):
        letter += 1
    elif n == " ":
        word += 1
    elif n == "." or n == "!" or n == "?":
        sentence += 1

# calculate index
grade = round(0.0588 * (letter / word * 100) - 0.296 * (sentence / word * 100) - 15.8)
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade " + str(grade))

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main(void)
{
    //prompt text
    string text = get_string("Text: ");
    
    //count letters, words, and sentences
    int l = 0;
    int w = 0;
    int s = 0;
    
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            l++;
        }
        else if (text[i] == ' ')
        {
            w++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            s++;
        }
    }
    w += 1;
    
    //calculate reading level
    float L = ((float) l / w) * 100;
    float S = ((float) s / w) * 100;
    float r = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(r);
    
    //print out grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}